src/es6/SirenUtil.js

```diff
@@ -0,0 +1,198 @@
+import 'd2l-fetch/d2l-fetch.js';
+import 'd2l-polymer-siren-behaviors/store/entity-store.js';
+import 'd2l-polymer-siren-behaviors/store/action-queue.js';
+import SirenParse from 'siren-parser';
+
+const _getSirenFields = function(action) {
+	const url = new URL(action.href, window.location.origin);
+	const fields = [];
+	if (action.method === 'GET' || action.method === 'HEAD') {
+		for (const param in url.searchParams.entries()) {
+			fields.push({ name: param[0], value: param[1] });
+		}
+		// Disable URLSearchParams until they are fully supported (i.e. Edge)
+		/*
+		} else if (window.URLSearchParams && action.type === 'application/x-www-form-urlencoded') {
```

Comment:
prob should just delete this code or do capability testing if we wanted to use it and gracefully fallback